### PR TITLE
Dont save whatsapp messages if validation fails 

### PR DESCRIPTION
**Pull request by megh-k**
Adding a validation to contentpage to ensure whatsapp message with attachment is shorter than 1024 chars

A `clean` method on the `ContentPage` object validates the whatsapp block to ensure that a message with an image, document or media cannot exceed 1024 characters (tested and python counts emoji's as 1 character)

An error banner is raised at the top with a list of errors - I was unable to get the errors onto a block level

whatsapp message validation failed

### CODE REVIEW

File: home/models.py [hunk 1]
```diff
@@ -340,6 +341,29 @@ def whatsapp_template_name(self):
...
 
+    def clean(self):
+        message_with_media_length = 1024
+        errors = []
+        for message in self.whatsapp_body:
+            if (
+                (
+                    "image" in message.value
+                    and "document" in message.value
+                    and "media" in message.value
```
Comment (rudigiesler) [inline home/models.py:352]: Will all these fields be in `value`, even if there's only a `document`, will the `image` and `media` fields also be present?

File: home/models.py [hunk 2]
```diff
@@ -340,6 +341,29 @@ def whatsapp_template_name(self):
...
+                    or message.value["document"]
+                    or message.value["media"]
+                )
+                and len(message.value["message"]) > message_with_media_length
+            ):
+                errors.append(
+                    f"A WhatsApp message with media cannot be longer than {message_with_media_length} characters long, your message is {len(message.value['message'])} characters long"
+                )
+        if errors:
+            raise ValidationError(errors)
```
Comment (rudigiesler) [inline home/models.py:365]: Should we maybe have a unit test, to ensure that the error gets raised when we expect it to?

File: home/models.py (hunk 1 above)
Comment (megh-k) [reply home/models.py:352]: When the import PR is merged in, I would like to take this out, at the moment the import doesnt create objects under those 3 keywords, it only creates for `"image"` but if `"image"` is present in the block and the other two arent, the clean fails. This is just a work-around for the tests and using the import. But pages created by hand, and pages created with the updated import will have all three objects in the block

File: home/models.py (hunk 2 above)
Comment (megh-k) [reply home/models.py:365]: yes we should probably have unit tests for everything :D

**Review by codiebeulaine — APPROVED** [review]

**Pull request merged**